### PR TITLE
Add updates to guides

### PR DESCRIPTION
**Pull request by prokarius**
Addresses #141

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -621,6 +620,31 @@ image::HideInfoSequenceDiagrams.png[]
...
 
+=== [Proposed] Improved `delete` functionality
+
+==== Current Implementation
+When the user deletes a loan from the LoanBook, it deletes the correct loan, then commits the result into memory. What this entails is that the entire XML file gets overwritten when each delete command is being called.
+
+==== Proposed implementation:
+From above, there is already a `LoanStatus` that has been implemented. What this does is that it allows the loan to be marked as deleted by setting the enum to the appropriate value. This would result in the LoanBook not needing to recommit every change.
+
+From here, when the user decides to close the LoanBook, it will then loop through all the loans and filter out the ones which have no been delete. These are the loans that would be kept and saved in the LoanBook.
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: "...which have not been deleted."

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -621,6 +620,31 @@ image::HideInfoSequenceDiagrams.png[]
...
+** Pros: Easy to implement
+** Cons: Large time taken for memory operations as reading and writing to the data is required.
+
+* Alternative 2 (proposed implementation): Using an enum to implement lazy deletion.
+** Pros: Much more time efficient due to lazy deletion
+** Cons: Implementation is harder, and changing to this implementation would affect the undo and redo features.
+
+* Alternative 3: Fix the LoanBook storage manager so that it does not rewrite the entire XML file after each delete operation.
+** Pros: All operations are now lazy. Much better performance of the application because commit operations do not need to change the large XML file.
+** Cons: Large refactoring work required. May potentially affect the timeline of development of the other functions.
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: Inconsistent tense in the titles for the alternatives. Try:
1) Remove each loan as they are deleted.
2) Use an enum to implement lazy deletion.
(3 is fine.)

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -768,6 +792,21 @@ e.g. `loanbook.storage.StorageManagerTest`
 ..  Hybrids of unit and integration tests. These test are checking multiple code units as well as how the are connected together. +
 e.g. `loanbook.logic.LogicManagerTest`
 
+[WARNING]
+====
+The tests within the `add` command parsers do not
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: Do not what?

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -1226,6 +1265,75 @@ _{ more test cases ... }_
...
+
+.. Choose a loan that has not been returned. These loans are marked with the ONGOING tag. Find the index number of it and run `return i/<INDEX>`. +
+   Expected: The loan is returned successfully. The display message details that the loan has been deleted, and gives a suggestion for the cost price of the duration of the loan.
+
+.. Choose another ongoing loan from the list and run `return i/<INDEX>`. +
+   Expected: The loan is returned successfully as well. The display message details that the loan has been deleted, and gives a suggestion for the cost price of the duration of the loan.
+
+. Negative tests:
+.. Test case: `return i/1` +
+   Expected: First loan from the list is marked as returned. Details of the returned loan shown in the status message. Timestamp in the status bar is updated. The values of the card on the left of the application has been changed to reflect the returning of the loan.
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: This is not a negative test case.

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -1226,6 +1265,75 @@ _{ more test cases ... }_
...
+
+.. Test case: `return i/0` +
+   Expected: No loan is returned. Error details shown in the status message. Status bar remains the same.
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: Could be more specific on what error details will be seen.

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -1226,6 +1265,75 @@ _{ more test cases ... }_
...
+.. Other incorrect `return` commands to try:
+... `return`
+... `return x` (where `x` is larger than the list size)
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: Shouldn't it be `return i/x`? Could also mention that `x` is an integer, just to be precise.

Speaking of which, some of the other commands (e.g. `edit` and the old `delete` iirc) that take in an index don't bother using the `i/` header, especially if that's the only argument. Something for consideration in the implementation.

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -1226,6 +1265,75 @@ _{ more test cases ... }_
...
+... `return i/aaa` (Where `aaa` is a string)
+... `return y` (where `y` is a loan that has already returned). +
```
Comment (FongYuan) [inline docs/DeveloperGuide.adoc]: "`y` is the index of a loan that has already been returned."
If you're planning to add more loan statuses, it may be helpful to rephrase this as "`y` is the index of a loan that is not `Ongoing`."

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -73,26 +85,33 @@ Format: `list`
...
-* `edit i/90210 p/James Tan hp/86661352 e/alabaaoe@xyz.com r/144` +
-For the loan with Loan ID 90210, changes the customer to James Tan, who has the specified contact details, to the rate of $144 per hour.
+* `edit i/11 o/Bicycle-017` +
+For the loan at index 11, changes the loaned object from whatever it was before to Bicycle-017.
+* `edit i/9 p/James Tan hp/86661352 e/alabaaoe@xyz.com r/144` +
+For the loan at index 9, changes the customer to James Tan, who has the specified contact details, to the rate of $144 per hour.
+
+[NOTE]
+====
+We do not allow editing of the loan's start time or the return time because we want you to be able to use our application as an auditing tool. We hope that this would simplify your business. However, by allowing you to edit the start and end time, this would invalidate this application as an auditing tool.
```
Comment (FongYuan) [inline docs/UserGuide.adoc]: The second sentence just repeats the previous point. Redundant.

File: docs/UserGuide.adoc [hunk 9]
```diff
@@ -73,26 +85,33 @@ Format: `list`
...
+
+We are currently looking into alternatives that would allow you to edit the start and end time though! Stay tuned for our announcement in LoanBook version 2.0!
```
Comment (FongYuan) [inline docs/UserGuide.adoc]: Sounds like it contradicts your stance in the previous paragraph.

File: docs/UserGuide.adoc [hunk 10]
```diff
@@ -149,13 +168,35 @@ Set the password of the app to `n3wP4sS`.
 
 === Returning a loan : `return`
 
-Marks a loan as returned based on LOAN_ID and automatically prints out the amount payable. +
-Format: `return i/LOAN_ID`
+What's the use of a LoanBook if you are unable to return the loans that you have? This is how you can do that! In fact, we see your pains trying to calculate the amount payable on a calculator, so we decided to help you out by doing all the number crunching for you.
+
+Marks a loan as returned based on LOAN_INDEX and automatically prints out the amount payable. The amount payable will be prorated and rounded down to the nearest minute. +
```
Comment (FongYuan) [inline docs/UserGuide.adoc]: Missing subject: "This command marks a loan as returned..."

File: docs/UserGuide.adoc [hunk 11]
```diff
@@ -174,6 +215,8 @@ The difference between deleting all the loans and hard resetting the LoanBook is
 
 === Summarize all the transactions : `summary`
 
+Do you want to find out at a glance how much money your business has collected? Or maybe out of all the bicycles you have, how many are currently loaned out? Here's the command for you.
+
 Show the total number of loans that are done and in progress. Also summarizes the loan status of each item, the number of times an object was loaned before.
```
Comment (FongYuan) [inline docs/UserGuide.adoc]: Same as above: "This command shows the total number..."

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (prokarius) [reply docs/DeveloperGuide.adoc]: Good catch. That's weird though, I think I failed to copy and paste one of the lines from my draft. Thanks for pointing this out!

File: docs/DeveloperGuide.adoc (hunk 4 above)
Comment (prokarius) [reply docs/DeveloperGuide.adoc]: I have removed that test case, as it was a remnant of a draft I had before. Thanks.

File: docs/UserGuide.adoc (hunk 9 above)
Comment (prokarius) [reply docs/UserGuide.adoc]: From what I was thinking, there could be other alternatives where this is available. Perhaps it may be better to not mention this as a potential future feature.

File: docs/DeveloperGuide.adoc (hunk 6 above)
Comment (prokarius) [reply docs/DeveloperGuide.adoc]: Yes it should be. I plan to use the `i/` header because in future development, I would like to allow users to return a loan based on other parameters that are not just the index of the loan.
